### DISCUSSION TITLE
My rice for small 10 inches laptop

### DISCUSSION THREAD

**Discussion by RiccardoPP** [labwc/labwc#2151] [Show and tell]
![20240916_10h56m44s_grim](https://github.com/user-attachments/assets/5306c832-7049-4a8a-9350-2be95e6f0641)

Recently I've restored an old EeePC 1025C with Intel Cedar Trail CPU (and the ugly PowerVR GPU...) and crafted a rice to use as secondary (or travel) little machine. Great battery life, a superb matte display and good performance with Alpine Linux (it could be better if it run entirely from RAM). There is a video on my YT (in Italian): https://youtu.be/-sMsHLxLWgs if you like it.

Dotfiles: https://github.com/RiccardoPP/dotfiles/tree/main/LabWC-Alpine-Netbook

Some info:

- Laptop: Asus EeePC 1025C (N2800/3GB RAM/120GB SSD)
- Distro: Alpine Linux 32-bit (edge, community + testing repo)
- WM: LabWC 0.8.0
- Terminal: Foot
- Bar: Waybar
- Launcher: Tofi
- Editor: Helix / Micro
- File Manager: PCManFM
- Fonts: Misc Ohsnap / Terminus / Font Awesome
- Icons: Haiku

Thanks for LabWC!

![20240914_11h53m18s_grim](https://github.com/user-attachments/assets/66133880-300b-4887-a035-f111d3930e74)
![20240916_10h57m00s_grim](https://github.com/user-attachments/assets/67d129ce-6f9c-4d74-92e5-ceb951f41fc9)
![20240916_10h59m40s_grim](https://github.com/user-attachments/assets/7ae1346e-8fb9-4ef2-976d-f3f0220b7aae)
![20240916_10h58m55s_grim](https://github.com/user-attachments/assets/ead32b54-493a-40f0-a097-7cf77b27e3cc)
![20240916_10h59m20s_grim](https://github.com/user-attachments/assets/f2f717cb-e2a1-4be5-a808-afa5ab68c816)
![20240916_10h58m04s_grim](https://github.com/user-attachments/assets/d4172e89-cf50-421c-87ec-87908d06698e)

**Comment by cerement** [comment]
now that you’ve put all that work into it, it’s also possible to [replace your bars/panels with notifications](https://www.youtube.com/watch?v=LbG_a3drzNE) in order to reclaim a few more pixels of screen real estate

**Reply by RiccardoPP** [reply]
Yes, I've used a similar solution years ago (this one, for example https://riccardo.im/articoli/restauro-portatile-linux/), or simply putting waybar with transparent background in a bottom layer in the desk, but now I like to have the bar visible :)

**Comment by johanmalm** [comment]
Absolutely awesome. Great work!
I love the video too, although don't understand the language.

**Reply by RiccardoPP** [reply]
Thanks Johan.

**Reply by 01micko** [reply]
Love it too! 'LabWC' sounds better in Italian. :it:

**Reply by glow12121** [reply]
I used to have EeePC 1025c years ago. My reliable friend in difficult times. Thanks for sharing the settings!

**Comment by lobre** [comment]
Hey, late to the party, but this is a nice rice!

I was wondering how you achieve that "dot" at the left of the titlebar. Is it a global icon for all apps, or kind of a button that has a "dot" shape?

And another question, what is this "Customize Look and Feel" application?

**Reply by RiccardoPP** [reply]
Hello, check my dotfiles: https://github.com/RiccardoPP/dotfiles/tree/main/LabWC-Alpine-Netbook

README is in Italian but you could translate it or just download the files. 

"Customize Look and Feel" is lxappearance but if you are in Arch, Debian or other distro than Alpine Linux, use nwg-look to customize that things.

**Comment by johanmalm** [comment]
I thought I recognised `lxappearance` there.

The dot is the default menu button until a couple version ago.

Ref:
- https://github.com/labwc/labwc/commit/6117e85820f1d3c83ef71f3bd8b8f5c6c483be7c

**Reply by Consolatis** [reply]
If you want to restore the old dot this should likely work (not tested though). Save as `$your_theme_directory/openbox-3/menu.xbm`:
```c
#define menu_width 6
#define menu_height 6
static unsigned char menu_bits[] = {
   0x00, 0x18, 0x3c, 0x3c, 0x18, 0x00 };
```

**Reply by lobre** [reply]
Thanks a lot!

For other people who would also search for that, apart from what @Consolatis said, you need the `menu` in the layout of your titlebar inside of `rc.xml`. Example below.

```
<layout>menu,close:max</layout>
```

Last general question, do you know why the default icons are in `xbm` format instead of say `svg` (which seems better in terms of scaling)? Is it to keep it simple, or because `xbm` is still recommended as the good approach, or maybe to stay close to Openbox?

I am implementing my own theme and wondering if I should stay with `xbm` or switch to `svg`.

**Reply by Consolatis** [reply]
SVG support is technically optional in labwc. So using that for default icons would cause issues when labwc is compiled without librsvg.

If you develop your own theme you can use it though (e.g. `menu-active.svg` + `menu-inactive.svg` and maybe the two additional `menu_hover` variants). I don't remember the lookup priority on the top of my head though so you may need to rename / move the `menu.xbm` icon.

**Reply by lobre** [reply]
Understood, thanks for confirming!
